linera-service/src/storage.rs

```diff
@@ -110,6 +121,31 @@ impl FromStr for StorageConfig {
             let namespace = s.to_string();
             return Ok(Self::Memory { namespace });
         }
+        if let Some(s) = input.strip_prefix(STORAGE_SERVICE) {
+            if s.is_empty() {
+                return Err(format_err!(
+                    "For Storage service, the formatting has to be service:endpoint:namespace"
+                ));
+            }
+            let parts = s.split(':').collect::<Vec<_>>();
+            if parts.len() == 4 {
+                let protocol = parts[0];
+                let address = parts[1];
+                let port = parts[2];
+                let mut endpoint = protocol.to_string();
+                endpoint.push(':');
+                endpoint.push_str(address);
+                endpoint.push(':');
+                endpoint.push_str(port);
+                let endpoint = endpoint.to_string();
+                let namespace = parts[3].to_string();
+                return Ok(Self::Service {
+                    endpoint,
+                    namespace,
+                });
+            }
+            return Err(format_err!("We should have one endpoint and one namespace"));
```

Comment:
Would be easier to read if this were returned early, if `parts.len() != 4`. Or even better with `ensure!`.

Reply:
Alternatively, you could use `s.rsplitn(2, ':')`, then you wouldn't have to put the endpoint back together.

Reply:
Ok, I took the first option.